lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -922,19 +922,21 @@ void SwiftASTContext::ScopedDiagnostics::PrintDiagnostics(
                              last_line);
 }
 
-bool SwiftASTContext::ScopedDiagnostics::HasErrors() const {
+std::optional<SwiftASTContext::ScopedDiagnostics::ErrorKind>
+SwiftASTContext::ScopedDiagnostics::GetErrorKindIfAny() const {
+  using ErrorKind = SwiftASTContext::ScopedDiagnostics::ErrorKind;
   auto &consumer = *static_cast<StoringDiagnosticConsumer *>(&m_consumer);
 
   if (consumer.m_num_swift_errors > m_cursor.m_num_swift_errors)
-    return true;
+    return ErrorKind::swift_error;
   if (consumer.m_raw_clang_diagnostics.size() > m_cursor.clang)
-    return true;
+    return ErrorKind::clang_error;
```

Comment:
Is it guaranteed that these are never both true? In other words should the result be a bit mask?

Reply:
I'm not sure if they're exclusive (probably not), but I don't there's any advantage in reporting both of them at the same time right now. What if we prefer reporting the clang error before the swift one?

Reply:
@kastiglione I reordered the swift and clang errors to report the clang one first, do let me know if you think it's worth changing this to a bit mask though

Reply:
for the current use, putting clang first is a good call.

It does seem like there could be a future confusion that `clang` means no `swift` errors, or vice versa. But that can be dealt with if it happens.